ql/src/java/org/apache/hadoop/hive/ql/metadata/HiveStorageHandler.java

```diff
@@ -266,7 +267,7 @@ default boolean canProvideBasicStatistics() {
    * @return boolean
    */
   default boolean setColStatistics(org.apache.hadoop.hive.ql.metadata.Table table,
-      List<ColumnStatistics> colStats) {
+      List<ColumnStatistics> colStats, ColumnStatsDesc columnStatisticsDesc) {
```

Comment:
we can't just change the public API by decommissioning an older signature

Reply:
It was an oversight, we will need to modify it. We will need the columnStatisticsDesc for the next improvements as well. Since it is not out in any of the hive releases yet would it be ok to modify the signature?

Reply:
Since we are changing the API, can we think of returning a map<Column, Boolean Result> OR a generic object containing mapping of (Column Name -> Merge result) (since we cant predict things that this generic object will contain).
This return type is useful in the case of Alter table Change column's datatype.

@simhadri-g @deniskuzZ WDYT?

Reply:
> @SourabhBadhya, why do we need per column Merge result?
@deniskuzZ If there is a query which does Alter table Change column datatype.

The column stats for that given column written by the older snapshots is of a different type. Hence they cant be merged. But other columns can proceed with merging the column stats (at least this seems to be the current behaviour for native tables).

Reply:
The mergeColStats  from the MetaStoreServerUtils is a void method.   https://github.com/apache/hive/blob/master/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/utils/MetaStoreServerUtils.java#L795

Inorder to return (Column Name -> Merge result )  we will have to modify the MetaStoreServerUtils.mergeColStats to return the result for each column or move the set column stats accurate to MetaStoreServerUtils.mergeColStats method.

Since this would require a lot of changes, can we address that in a different jira?

Reply:
sure, added the same comment above, it seems you don't use columnStatisticsDesc obj